clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3241,6 +3241,73 @@ parameter.
   }];
 }
 
+def SYCLIntelFPGAIpInterfaceCsrDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "intel::ip_interface_csr";
+  let Content = [{
+``[[intel::ip_interface_csr]]`` attribute applies to a function/lambda function.
+Takes no arguments and applies to the IP component that uses CSR
+(Control-Status Registers) invocation interface. Cannot be applied multiple times
+to the same function. Cannot be combined with``[[intel::ip_interface_streaming]]``
+or ``[[intel::ip_stall_free_return]]``.
+
+if ``intel::ip_interface_csr`` is applied to a function called from a
+device kernel, the attribute is ignored and it is not propagated to the kernel.
```

Comment:
I've read all of these words a few times now and I have no idea when to use this attribute, what it actually does, or why this needs to be an attribute in the first place.

It sounds a bit like this is telling the compiler about some information that it should already be able to figure out for itself. Is this attribute necessary? 

Why can this attribute not be applied multiple times? What harm is that preventing?